game/scripts/vscripts/abilities/sohei/sohei_palm_of_life.lua

```diff
@@ -0,0 +1,233 @@
+
+--------------------------------------------------------------------------------
+
+function modifier_sohei_palm_of_life_movement:IsDebuff()
+	return false
+end
+
+function modifier_sohei_palm_of_life_movement:IsHidden()
+	return true
+end
+
+function modifier_sohei_palm_of_life_movement:IsPurgable()
+	return false
+end
+
+function modifier_sohei_palm_of_life_movement:IsStunDebuff()
+	return false
+end
+
+function modifier_sohei_palm_of_life_movement:GetPriority()
+	return DOTA_MOTION_CONTROLLER_PRIORITY_MEDIUM
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_sohei_palm_of_life_movement:CheckState()
+	local state = {
+		[MODIFIER_STATE_NO_UNIT_COLLISION] = true,
+	}
+
+	return state
+end
+
+--------------------------------------------------------------------------------
+
+if IsServer() then
+	function modifier_sohei_palm_of_life_movement:OnCreated( event )
+		-- Movement parameters
+		local parent = self:GetParent()
+		self.target = EntIndexToHScript( event.target )
+		self.speed = event.speed
+		self.tree_radius = event.tree_radius
+		self.endDistance = event.endDistance
+		self.doHeal = event.doHeal > 0
+
+		if self:ApplyHorizontalMotionController() == false then
+			self:Destroy()
+			return
+		end
+
+		-- Trail particle
+		local trail_pfx = ParticleManager:CreateParticle( "particles/econ/items/juggernaut/bladekeeper_omnislash/_dc_juggernaut_omni_slash_trail.vpcf", PATTACH_CUSTOMORIGIN, parent )
+		ParticleManager:SetParticleControl( trail_pfx, 0, parent:GetAbsOrigin() )
+		ParticleManager:SetParticleControl( trail_pfx, 1, (  self.target:GetAbsOrigin() - parent:GetAbsOrigin() ):Normalized() * 300 )
+		ParticleManager:ReleaseParticleIndex( trail_pfx )
+	end
+
+--------------------------------------------------------------------------------
+
+	function modifier_sohei_palm_of_life_movement:OnDestroy()
+		local parent = self:GetParent()
+
+		parent:FadeGesture( ACT_DOTA_RUN )
+		parent:RemoveHorizontalMotionController( self )
+		ResolveNPCPositions( parent:GetAbsOrigin(), 128 )
+	end
+
+--------------------------------------------------------------------------------
+
+	function modifier_sohei_palm_of_life_movement:UpdateHorizontalMotion( parent, deltaTime )
+		local parentOrigin = parent:GetAbsOrigin()
+		local targetOrigin = self.target:GetAbsOrigin()
+		local dA = parentOrigin
+		dA.z = 0
+		local dB = targetOrigin
+		dB.z = 0
+		local direction = ( dB - dA ):Normalized()
+
+		local tickSpeed = self.speed * deltaTime
+		tickSpeed = math.min( tickSpeed, self.endDistance )
+		local tickOrigin = parentOrigin + ( tickSpeed * direction )
+
+		parent:SetAbsOrigin( tickOrigin )
+		parent:FaceTowards( targetOrigin )
+
+		GridNav:DestroyTreesAroundPoint( tickOrigin, self.tree_radius, false )
+
+		local distance = parent:GetRangeToUnit( self.target )
+
+		if distance <= self.endDistance then
+			if self.doHeal then
+				-- do the heal
```

Comment:
Can you throw a sound event for the heal in there too, and we'll find one to fit it